### PR TITLE
✨ UI indicating that the worker is workering on a document

### PR DESCRIPTION
**Pull request by anuejn**
this should allow us to indecate that work is currently done on a document in the frontend

### CODE REVIEW

**Comment by anuejn** [comment]
I think this is good to merge now since it is better than nothing even if it might not be the final UI yet

File: frontend/src/editor/worker_status.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                    stroke={getColor(data[dependencyIndex])}
+                  />
+                );
+              })}
+            </>
+          );
+        })}
+        {data?.map((task, i) => {
+          const ourY = yPositionsCircles[i];
+          console.log(task);
```
Comment (pajowu) [inline frontend/src/editor/worker_status.tsx:93]: 🖨️

File: frontend/src/editor/worker_status.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+          const ourY = yPositionsCircles[i];
+
+          return (
+            <>
+              {task?.dependency?.map((dependency) => {
+                const dependencyIndex = data?.findIndex((task) => task?.id == dependency);
+                const dependencyY = yPositionsCircles[dependencyIndex];
+
+                const curvature = (i - dependencyIndex) * 10;
+                console.log(data[dependencyIndex].task_type, task.task_type, i - dependencyIndex);
```
Comment (pajowu) [inline frontend/src/editor/worker_status.tsx:70]: 🖨️

**Review by pajowu — APPROVED** [review]

**Comment by pajowu** [comment]
I'm a big fan of the animation 🌈

**Pull request merged**